### PR TITLE
moar restructuring

### PR DESCRIPTION
**Pull request by huwaireb**
partially completes #28. I just extracted the Typst part onto `crates/compiler`

Chiitan Car Push

### CODE REVIEW

File: Justfile [hunk 1]
```diff
@@ -33,19 +33,7 @@ app-format:
...
-scraper-lint:
-        cargo clippy -p scraper
-
-[group('scraper')]
-scraper-run:
-        cargo run -p scraper
-
-[group('scraper')]
-scraper-build:
-        cargo build -p scraper
```
Comment (DieracDelta) [inline Justfile:48]: I see that scraper was renamed to dataset. But why delete the targets instead of rename?

**Review by DieracDelta — CHANGES_REQUESTED** [review]
LGTM barring a nit in the justfile. Nice!

File: Justfile (hunk 1 above)
Comment (DieracDelta) [reply Justfile:48]: I think we also probably need targets for the other workspace members.

Comment (huwaireb) [reply Justfile:48]: discussed on discord:

Given `crates/dataset` will act in the future as a library it should not have it's own target nor Nix package. 
Such should be reserved for CLIs e.g a training cli & the actual app cli

but we added a package agnostic cargo-build target for use as ``just cargo-build package-name``

**Review by DieracDelta — APPROVED** [review]

**Pull request merged**